app/src/androidTest/java/org/simple/clinic/bp/BloodPressureRepositoryAndroidTest.kt

```diff
@@ -127,6 +128,41 @@ class BloodPressureRepositoryAndroidTest {
     assertThat(savedBloodPressure).isEqualTo(expected)
   }
 
+  @Test
+  fun observing_a_deleted_blood_pressure_as_a_stream_should_work_correctly() {
+    val now = Instant.now(clock)
+    val bloodPressure = testData.bloodPressureMeasurement(
+        createdAt = now,
+        updatedAt = now,
+        deletedAt = null,
+        syncStatus = SyncStatus.DONE)
+    appDatabase.bloodPressureDao().save(listOf(bloodPressure))
+
+    val deletedMeasurementObserver = repository.deletedMeasurementAsStream(bloodPressure.uuid)
+        .test()
+
+    val fifteenSeconds = Duration.ofSeconds(15L)
+    val tenHours = Duration.ofHours(10L)
+    val twentyDays = Duration.ofDays(20L)
+
+    listOf(fifteenSeconds, tenHours, twentyDays)
+        .map { now.plus(it) }
+        .map { bloodPressure.copy(deletedAt = it) }
+        .forEach { appDatabase.bloodPressureDao().save(listOf(it)) }
```

Comment:
Wanna remove `forEach`?

```kotlin
    listOf(fifteenSeconds, tenHours, twentyDays)
        .map { now.plus(it) }
        .map { bloodPressure.copy(deletedAt = it) }
        .apply {
          appDatabase.bloodPressureDao().save(it)
        }
```

Reply:
Techincally, this is the wrong usage of `.apply` since that is meant to make things like setting parameters on an object easier.

But I always want to simulate the blood pressure emitting multiple change notifications so I have to save them individually instead of as a single list.